yaksh/forms.py

```diff
@@ -184,3 +184,28 @@ class CourseForm(forms.ModelForm):
     class Meta:
         model = Course
         fields = ['name', 'active', 'enrollment']
+
+class EditProfile(forms.ModelForm):
+    """ edit profile form for students and moderators """
+
+    class Meta:
+        model = Profile
+        fields = ['first_name', 'last_name', 'institute', 'department', 'roll_number', 'position']
+
+    first_name = forms.CharField(max_length=30)
+    last_name = forms.CharField(max_length=30)
+
+    def __init__(self, user, *args, **kwargs):
+        initial = ''
+        if 'instance' in kwargs:
+            initial = kwargs["instance"]
+        if 'user' in kwargs:
+            user = kwargs["user"]
+            del kwargs["user"]
+        super(EditProfile, self).__init__(*args, **kwargs)
+        self.fields['first_name'].initial = user.first_name
+        self.fields['last_name'].initial = user.last_name
+        self.fields['roll_number'].initial = user.profile.roll_number
```

Comment:
If you are using model forms then you do not need to initialize fields after last_name,
pass profile instance to the form